WordPress/WordPressApi/WordPressComOAuthClient.swift

```diff
@@ -0,0 +1,192 @@
+    */
+    public func requestOneTimeCodeWithUsername(username: String, password:String,
+                                        success: () -> (), failure: (error: NSError) -> ())
+    {
+        let parameters = [
+            "username": username,
+            "password": password,
+            "grant_type": "password",
+            "client_id": clientID,
+            "client_secret": secret,
+            "wpcom_supports_2fa": true,
+            "wpcom_resend_otp": true
+        ]
+
+        sessionManager.POST("token", parameters:parameters, success:{ (task, responseObject) in
+            success()
+            }, failure: { (task, error) in
+                guard let httpURLResponse = task?.response as? NSHTTPURLResponse
+                else {
+                    failure(error: error)
+                    return
+                }
+                let processedError = self.processError(error, response:httpURLResponse)
+                // SORRY:
+                // SMS Requests will still return WordPressComOAuthErrorNeedsMultifactorCode. In which case,
+                // we should hit the success callback.
+                if processedError.code == WordPressComOAuthError.NeedsMultifactorCode.rawValue {
+                    success()
+                } else {
+                    failure(error: error)
+                }
+        })
+    }
+
+    private func processError(error: NSError, response:NSHTTPURLResponse) -> NSError {
+        guard response.statusCode >= 400 && response.statusCode < 500 && error.domain == AFURLResponseSerializationErrorDomain,
+        let responseData = error.userInfo[AFNetworkingOperationFailingURLResponseDataErrorKey] as? NSData,
+        let responseDictionary = (try? NSJSONSerialization.JSONObjectWithData(responseData, options:NSJSONReadingOptions())) as? [String:AnyObject],
+        let errorCode = responseDictionary["error"] as? String,
+        let errorDescription = responseDictionary["error_description"] as? String
+        else {
+            return error
+        }
+        /*
+         Possible errors:
+         - invalid_client: client_id is missing or wrong, it shouldn't happen
+         - unsupported_grant_type: client_id doesn't support password grants
+         - invalid_request: A required field is missing/malformed
+         - invalid_request: Authentication failed
+         - needs_2fa: Multifactor Authentication code is required
+         */
+
+        let errorsMap = [
+            "invalid_client" : WordPressComOAuthError.InvalidClient,
+            "unsupported_grant_type" : WordPressComOAuthError.UnsupportedGrantType,
+            "invalid_request" : WordPressComOAuthError.InvalidRequest,
+            "needs_2fa" : WordPressComOAuthError.NeedsMultifactorCode
+        ]
+
+        let mappedCode = errorsMap[errorCode]?.rawValue ?? WordPressComOAuthError.Unknown.rawValue;
+
+        return NSError(domain:self.dynamicType.WordPressComOAuthErrorDomain,
```

Comment:
I think it's a best practice in Swift to use the type name directly, unless you really need the dynamic type.